lib/ansible/plugins/inventory/docker_swarm.py

```diff
@@ -181,6 +181,8 @@ def _populate(self):
                     self.inventory.set_variable(self.node_attrs['ID'], 'docker_swarm_node_attributes', self.node_attrs)
                 if 'ManagerStatus' in self.node_attrs:
                     if self.node_attrs['ManagerStatus'].get('Leader'):
+                        self.inventory.set_variable(self.node_attrs['ID'], 'ansible_host',
```

Comment:
You should add a comment that this is necessary because of a bug in docker/moby. Otherwise, someone in the future could be confused why this is added and remove it (because "it works" for that person).

Reply:
Good point